lib/shape.ml

```diff
@@ -3,63 +3,76 @@ open Graphics
 type point = { x : int; y : int }
 type rectangle = { c : point; length : int; width : int }
 type circle = { c : point; radius : int }
-type ellipse = {c : point; rx : int; ry: int}
+type ellipse = { c : point; rx : int; ry : int }
 type shape = Circle of circle | Rectangle of rectangle | Ellipse of ellipse
 type shapes = shape list
 
-let dimensions = ref {x = 500; y = 500}
-let set_dimensions x y = 
-  dimensions := {x;y} 
+let dimensions = ref { x = 500; y = 500 }
+let set_dimensions x y =
+  dimensions := { x; y }
 
-let canvas_mid = { x = (!dimensions.x / 2); y = (!dimensions.y / 2)}
+let canvas_center = ref { x = 0; y = 0 }
```

Comment:
Why is this a reference? Since we're not mutating it anywhere, this can be a plain point.

Reply:
Oh I'll correct that.